scripts/_common.sh

```
@@ -0,0 +1,688 @@
+  # call on_init if exists
+  fn_exists on_init && on_init "$@"
+
+  local only_args=0 # 1 is true, 0 is false
+  local args=()
+  if [[ "${1-}" == -?* ]]; then
+    only_args=1
+    local OPT
+    while get_options "$@"; do
+      if [[ "$OPT" == -- ]]; then
+        # add param to accumulated arguments
+        args+=($OPTARG)
```

Shellcheck (0.10.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)